### ISSUE TITLE
Named jobs

### ISSUE THREAD

**Issue by enahsor** [dbader/schedule#502]
I'm not sure if this is possible with the current API - I have not seen any documentation supporting this. Is there some way to provide a job with an identifier? eg. I schedule a job called upload-stuff and then I'd be able to refer to that job throughout the lifetime of the schedule. I think this would prove super useful. If you do not think so, please let me know why. If you do agree that this would be useful, I'm open to attempting to implement it.

**Comment by leirons** [comment]
Ye, i think it would be cool

**Comment by LuisHenri** [comment]
You can add tags to the Job: https://github.com/dbader/schedule/blob/8a944c845fb837545f990639b227c2bbfd2f53ba/schedule/__init__.py#L443

You can add multiple Tags, but if you add only one, you could use it as its name by using `next(iter(job.tags))`

**Comment by rocky4546** [comment]
Yep, I have been using tags for a while.  It has the ability to set and then find the tags.  My tags represent a task and then I can have multiple events associated with that tag and pull up all events with a query.  It works great.

**Comment by SijmenHuizenga** [comment]
Hi! Totally agree that it is useful to have a human readable name attached to a job. This is what [tags](https://schedule.readthedocs.io/en/stable/examples.html#get-several-jobs-filtered-by-tags) are intended for. Is there is anything specific that you would like to achieve that can't be done without tags?

**Comment by mdwybo** [comment]
Hi. Actually there is something I would like to achieve that could be done with tags but there is no means of doing so that I can find. I would like to get the name of the next job to run. This could be its unique tag. I simply want to retrieve either the tag of the next job or the function that it calls and display it along with the time of its scheduled execution. And to make it more interesting I would like to do this in CircuitPython. Is this possible ? Thanks in advance.

**Comment by mdwybo** [comment]
Hi. Actually there is something I would like to achieve that could be done with tags but there is no means of doing so that I can find. I would like to get the name of the next job to run. This could be its unique tag. I simply want to retrieve either the tag of the next job or the function that it calls and display it along with the time of its scheduled execution. And to make it more interesting I would like to do this in CircuitPython. Is this possible ? Thanks in advance.
